### ISSUE TITLE
Fatal error: Class 'PHPStan\PhpDocParser\Parser\TokenIterator' not found

### ISSUE THREAD

**Issue by jrfnl** [slevomat/coding-standard#695]
Following up on #632 which reported the same error and was unjustly dismissed. /cc @dereuromark

This *is* an issue with the Slevomat standard and should be solved here.

Setup:
* PHP 7.2.13
* SlevomatCS git clone with branch `master` checked out.
    - Ran `composer install --no-dev`.
    - As Phing is not available on no-dev, did not run `bin/phing`
* PHPCS git clone with branch `master` checked out.
    - `phpcs` command available globally.
    - Added the path to SlevomatCS to the PHPCS `installed_paths`.

When I run the following command on a random code directory (to test which sniffs I may want to add to the ruleset for the project):
`phpcs . --extensions=php --basepath=. --tab-width=4 --standard=SlevomatCodingStandard`

... I receive the following fatal error:
```bash
# phpcs . --extensions=php --basepath=. --tab-width=4 --standard=SlevomatCodingStandard
PHP 7.2.13 (cli) (built: Dec  5 2018 07:34:22) ( ZTS MSVC15 (Visual C++ 2017) x86 )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies

PHP_CodeSniffer version 3.5.0 (stable) by Squiz (http://www.squiz.net)

Fatal error: Uncaught Error: Class 'PHPStan\PhpDocParser\Parser\TokenIterator' not found in /path/to/slevomat/SlevomatCodingStandard/Helpers/AnnotationHelper.php:245
Stack trace:
#0 /path/to/slevomat/SlevomatCodingStandard/Helpers/AnnotationHelper.php(224): SlevomatCodingStandard\Helpers\AnnotationHelper::parseAnnotationContent('@param', 'string $ext Ext...')
#1 /path/to/slevomat/SlevomatCodingStandard/Sniffs/Commenting/DocCommentSpacingSniff.php(124): SlevomatCodingStandard\Helpers\AnnotationHelper::getAnnotations(Object(PHP_CodeSniffer\Files\LocalFile), 151)
#2 /path/to/PHP_CodeSniffer/src/Files/File.php(489): SlevomatCodingStandard\Sniffs\Commenting\DocCommentSpacingSniff->process(Object(PHP_CodeSniffer\Files\LocalFile), 151)
#3 /path/to/PHP_CodeSniffer/src/Files/LocalFile.php(91): PHP_CodeSniffer\Files\File->process()
#4 /path/to/PHP_CodeSniffer/src/Runner.php(630): PHP_CodeSniffer\Files\LocalFile->process()
#5 /path/to/PHP_CodeSniffer/src/Runner.php(434): PHP_CodeSniffer\Runner->processFile(Object(PHP_CodeSniffer\Files\LocalFile))
#6 /path/to/PHP_CodeSniffer/src/Runner.php(114): PHP_CodeSniffer\Runner->run()
#7 /path/to/PHP_CodeSniffer/bin/phpcs(18): PHP_CodeSniffer\Runner->runPHPCS()
#8 {main}
  thrown in /path/to/slevomat/SlevomatCodingStandard/Helpers/AnnotationHelper.php on line 245

Exit code is 255
```

The issue is really easy to reproduce:
1. Install the latest PHPCS Phar file: https://github.com/squizlabs/PHP_CodeSniffer/releases/download/3.4.2/phpcs.phar
2. Clone this repo and run `composer install --no-dev`.
3. Register the standard with PHPCS `path/to/phpcs.phar --config-set installed_paths /path/to/slevomat-clone`
4. Now from the root of this repo run: `path/to/phpcs.phar . --extensions=php --tab-width=4 --standard=SlevomatCodingStandard`

I've ran some tests:
* This happens on all release tags in the 5.x series.
* `git bisect` shows me that commit 824a235ae9889eb274b353346ae653adace4b3f4 is to blame.
* Running `composer install` (=dev) & `bin/phing` makes no difference. Problem still occurs.

Now, you can blame PHPCS's autoloader, but that is neither here nor there. The simple fact is that this is broken.

If PHPCS detects itself to be installed in the `vendor` directory, PHPCS's autoloader unloads the Composer autoload file to place itself _before_ it and then will re-register the Composer autoload file (if found) after its own autoloader, so the Composer autoload is still available.

PHPCS also offers the ability to add a custom autoloader if you'd need it.

Now, I've done some further analysis.

PHPCS offers four different ways to install the tool:
1. As a Phar.
2. Via PEAR.
3. As a git clone.
4. As a Composer dependency of project.

External standards can normally be installed via [3] or [4] and need to be registered with PHPCS via `installed_paths`.

What it comes down to is that the Slevomat standard will currently **_only_** work when PHPCS is installed via [4] and run via `vendor/bin/phpcs` in a project or `bin/phpcs` when in the Slevomat standard itself. 

**Running the Slevomat standard with any of the other (supported) ways to install PHPCS is currently broken** as the Slevomat standard expects the Composer `autoload.php` file to be included while for any of the other methods it won't automatically be.

This is not a shortcoming of PHPCS itself. This is an issue with the Slevomat standard which clearly never took any of the other install methods into account.

The solution is to register your own autoloader which will load the Composer autoload file if necessary.

**Comment by kukulich** [comment]
It still think it's bug of PHPCS but I'll merge your PR (after you fix issues) :)

**Comment by jrfnl** [comment]
> It still think it's bug of PHPCS

It's not. PHPCS will autoload the sniff files and - since 3.0.2 - other files in the `StandardName` directory. Anything else is _your_ responsibility. After all, how is PHPCS to know your directory structure and how to autoload _your_ other files ?
